runtime/src/accounts_index.rs

```diff
@@ -885,7 +872,27 @@ impl<T: 'static + Clone> AccountsIndex<T> {
         let mut w_roots_tracker = self.roots_tracker.write().unwrap();
         w_roots_tracker.roots.insert(slot);
         w_roots_tracker.uncleaned_roots.insert(slot);
-        w_roots_tracker.max_root = std::cmp::max(slot, w_roots_tracker.max_root);
+        // `AccountsDb::flush_accounts_cache()` relies on roots being added in order
+        assert!(slot >= w_roots_tracker.max_root);
+        w_roots_tracker.max_root = slot;
+    }
+
+    pub fn add_cache_root(&self, slot: Slot) {
```

Comment:
it this fn is really needed? The sole caller of this from `add_root` is doing redundant work with the above fn of `add_cache_root`.

Reply:
sublte difference, `add_cache_root` doesn't have the `w_roots_tracker.uncleaned_roots.insert(slot);` 😃

Reply:
then, I'd like the subtlty is controlled by a flag unifying them. the functions are so similar:

```rust
pub fn add_root(&self, slot: Slot, caching_enabled: bool) {
        let mut w_roots_tracker = self.roots_tracker.write().unwrap();
        w_roots_tracker.roots.insert(slot);
        // we delay cleaning until flushing!
        if !caching_enabled {
            w_roots_tracker.uncleaned_roots.insert(slot);
        }
        w_roots_tracker.max_root = std::cmp::max(slot, w_roots_tracker.max_root);
        // `AccountsDb::flush_accounts_cache()` relies on roots being added in order
        assert!(slot >= w_roots_tracker.max_root);
        w_roots_tracker.max_root = slot;
}
```

Reply:
hehe yeah that's much better, I got lazy and didn't want to change all the tests 😛